src/commands/common/send-telemetry-command.js

```diff
@@ -11,45 +10,41 @@ class SendTelemetryCommand extends Command {
         this.logger = ctx.logger;
         this.config = ctx.config;
-        this.telemetryInjectionService = ctx.telemetryInjectionService;
         this.networkModuleManager = ctx.networkModuleManager;
         this.blockchainModuleManager = ctx.blockchainModuleManager;
+        this.repositoryModuleManager = ctx.repositoryModuleManager;
+        this.telemetryModuleManager = ctx.telemetryModuleManager;
     }
 
     /**
      * Performs code update by fetching new code from github repo
      * @param command
      */
     async execute() {
-        if (!this.config.telemetry.enabled || !this.config.telemetry.sendTelemetryData) {
+        const implementationName = this.telemetryModuleManager.getImplementationNames();
+        if (
+            !this.config.modules.telemetry.enabled ||
+            !this.config.modules.telemetry.implementation[implementationName].config
+                .sendTelemetryData
+        ) {
             return Command.empty();
         }
+
         try {
-            const events = await this.telemetryInjectionService.getUnpublishedEvents();
-            const signalingMessage = {
-                nodeData: {
-                    version: pjson.version,
-                    identity: this.networkModuleManager.getPeerId().toB58String(),
-                    hostname: this.config.hostname,
-                    operational_wallet: this.blockchainModuleManager.getPublicKey(),
-                    management_wallet: this.blockchainModuleManager.getManagementKey(),
-                    triple_store: this.config.modules.tripleStore.defaultImplementation,
-                    auto_update_enabled: this.config.modules.autoUpdater.enabled,
-                    multiaddresses: this.networkModuleManager.getMultiaddrs(),
-                },
-                events: events || [],
-            };
-            const config = {
-                method: 'post',
-                url: this.config.telemetry.signalingServerUrl,
-                headers: {
-                    'Content-Type': 'application/json',
-                },
-                data: JSON.stringify(signalingMessage),
+            const events = (await this.getUnpublishedEvents()) || [];
+            const nodeData = {
+                version: pjson.version,
+                identity: this.networkModuleManager.getPeerId().toB58String(),
+                hostname: this.config.hostname,
+                operational_wallet: this.blockchainModuleManager.getPublicKey(),
+                management_wallet: this.blockchainModuleManager.getManagementKey(),
+                triple_store: this.config.modules.tripleStore.defaultImplementation,
+                auto_update_enabled: this.config.modules.autoUpdater.enabled,
+                multiaddresses: this.networkModuleManager.getMultiaddrs(),
             };
-            const response = await axios(config);
+            const response = await this.telemetryModuleManager.sendTelemetryData(nodeData, events);
```

Comment:
Since we only need information if telemetry data is sent or not (we don't need whole response object) update the method to return boolean.